File: Day321_2025/Q321_Leetcode.py
class Solution:
    def kLengthApart(self, nums, k: int) -> bool:
        prev = -1
        first = True

        for i, val in enumerate(nums):
            if first and val == 1:
                prev = i
                first = False
            elif val == 1:
                if i - prev - 1 < k:
                    return False
                prev = i
        
        return True

File: Day321_2025/Q321_Leetcode.java
class Q321_Leetcode {
    public boolean kLengthApart(int[] nums, int k) {
        int n = nums.length;
        int prev = -1; 
        boolean first = true;

        for (int i = 0; i < n; i++) {
            if (first && nums[i] == 1) {
                prev = i;
                first = false;
            } 
            else if (nums[i] == 1) {
                if (i - prev - 1 < k) return false;
                prev = i;
            }
        }
        return true;
    }
}
